pex/util.py

```diff
@@ -177,6 +177,9 @@ def cache_distribution(cls, zf, source, target_dir):
           with contextlib.closing(zf.open(name)) as zi:
             with safe_open(os.path.join(target_dir_tmp, target_name), 'wb') as fp:
               shutil.copyfileobj(zi, fp)
+              fileinfo = zf.getinfo(zi.name)
+              perm = fileinfo.external_attr >> 16
+              os.chmod(fp.name, int(str(perm), 8))
```

Comment:
Just `os.chmod(fp.name, fileinfo.external_attr >> 16)` - the ->str->int dance is not needed unless I'm missing something.